src/ModalFilterAdder.js

```diff
@@ -1,20 +1,14 @@
+import _ from 'lodash/fp'
 import FilterAdder from './FilterAdder'
-import { Modal, ModalPicker, NestedPicker } from './layout/'
+import { ModalPicker, NestedPicker } from './layout/'
 import { defaultProps } from 'recompose'
+import { defaultTheme } from './utils/hoc'
 
-export default ({
-  Item,
-  Input,
-  Button,
-  Highlight,
-  label = 'Add Custom Filter',
-  Picker = NestedPicker,
-} = {}) =>
-  defaultProps({
-    Picker: defaultProps({
-      Modal,
-      Button,
-      label,
-      Picker: defaultProps({ Input, Highlight, Item })(Picker),
-    })(ModalPicker),
-  })(FilterAdder)
+export default ({ label = 'Add Custom Filter', theme }) => {
+  let PickerForModalPicker = (theme && theme.Picker) || NestedPicker
+  let LabelledModalPicker = _.flow(
+    defaultTheme({ Picker: PickerForModalPicker }),
+    defaultProps({ label })
+  )(ModalPicker)
+  return defaultTheme({ Picker: LabelledModalPicker })(FilterAdder)
+}
```

Comment:
ModalFilterAdder in particular was completely broken by theme objects, so I had to rewrite it. This is a first pass - it doesn't throw React errors anymore, but still needs some love